src/content/docs/dev/general-message-passing/sui/gmp-tutorial.mdx

```diff
@@ -0,0 +1,918 @@
+
+You can then obtain the `modules` and `dependencies` from the compiled `buildOutput`.
+
+```js
+const { modules, dependencies } = JSON.parse(buildOutput);
+```
+
+At this point you should be able to run `node scripts/deploy.js` and build your function. The logs should appears as follows
+
+```bash
+INCLUDING DEPENDENCY GasService
+INCLUDING DEPENDENCY RelayerDiscovery
+INCLUDING DEPENDENCY AxelarGateway
+INCLUDING DEPENDENCY Utils
+INCLUDING DEPENDENCY VersionControl
+INCLUDING DEPENDENCY Sui
+INCLUDING DEPENDENCY MoveStdlib
+BUILDING gmp_example
+```
+
+### Access Wallet
+
+Before publishing to the blockchain you need to setup your wallet that will send the transaction.
+
+You can create a new `utils/index.js` file where you can define a new function called `getWallet()`
+
+This function will require several imports up top
+
+```js
+import { ethers } from 'ethers';
+import { Ed25519Keypair } from '@mysten/sui/keypairs/ed25519';
+import { SuiClient } from '@mysten/sui/client';
+import 'dotenv/config';
+```
+
+In the `getWallet()` function will create a new [Ed25519Keypair](https://sdk.mystenlabs.com/typedoc/classes/_mysten_sui.keypairs_ed25519.Ed25519Keypair.html) and `client` object that can be returned to be used later. The keypair will be created based on your private key that should be saved in your `.env` file.
+
+```js
+export function getWallet() {
+  const client = new SuiClient({ url: 'https://fullnode.testnet.sui.io:443' });
+
+  const rawKey = process.env.PRIVATE_KEY;
+  if (!rawKey) {
+    console.error('PRIVATE_KEY not set in your .env');
+    process.exit(1);
+  }
+
+  const keypair = Ed25519Keypair.fromSecretKey(rawKey);
+  return [keypair, client];
+}
+```
+
+Now back in your main `deploy.js` script in your `run()` function you can call `getWallet()` as follows
+
+```js
+  const [keypair, client] = getWallet()
+```
+
+
+### Publish
+To publish the package on chain you will need to sign a transaction from your Sui Wallet that deploys the package. 
+
+First import the `Transaction` from the Sui SDK 
+```js
+import { Transaction } from '@mysten/sui/transactions';
+```
+
+Then back in your `run()` function you can execute the transaction publishing your package.
+
+```js
+//create new tx
+const tx = new Transaction();
+//publish the package
+const [upgradeCap] = tx.publish({ modules, dependencies });
+```
+
+The `modules` and `dependencies` being passed are the same ones that were destructured from the build output above.
+
+The `upgradeCap` that is returned is an on-chain resource that the Sui mints for you whenever you publish a package, the address that holds the `cap` is able to upgrade the package at a future point. Further reading on this can be found [here](https://sdk.mystenlabs.com/typescript/transaction-building/basics#transactions). To ensure you are the one with the ability to upgrade the package that you deploy you can pass the `cap` to your own address. 
```

Comment:
```suggestion
The `upgradeCap` returned is an on-chain resource that the Sui mints for you whenever you publish a package; the address that holds the `cap` can upgrade the package at a future point. Further reading on this can be found [here](https://sdk.mystenlabs.com/typescript/transaction-building/basics#transactions). To ensure you are the one with the ability to upgrade the package that you deploy, you can pass the `cap` to your own address. 
```